shardingsphere-agent/shardingsphere-agent-core/src/main/java/org/apache/shardingsphere/agent/core/plugin/PluginAdviceDefine.java

```diff
@@ -0,0 +1,300 @@
+            this.classNameOfTarget = classNameOfTarget;
+        }
+
+        /**
+         * Intercept the new target.
+         *
+         * @param classNameOfTarget the class name of target.
+         * @return Configuration builder.
+         */
+        public Builder intercept(final String classNameOfTarget) {
+            // TODO not-implemented yet
+            return this;
+        }
+
+        /**
+         * to configure the intercepting point on constructor.
+         *
+         * @param matcher constraints
+         * @return Configuration builder
+         */
+        public ConstructorPointBuilder onConstructor(final ElementMatcher<? super MethodDescription> matcher) {
+            return new ConstructorPointBuilder(this, matcher);
+        }
+
+        /**
+         * to configure the intercepting point around instance method.
+         *
+         * @param matcher constraints
+         * @return Configuration builder
+         */
+        public InstanceMethodPointBuilder method(final ElementMatcher<? super MethodDescription> matcher) {
+            return new InstanceMethodPointBuilder(this, matcher);
+        }
+
+        /**
+         * to configure the intercepting point around instance method.
+         *
+         * @param matcher constraints
+         * @return Configuration builder
+         */
+        public StaticMethodPointBuilder staticMethod(final ElementMatcher<? super MethodDescription> matcher) {
+            return new StaticMethodPointBuilder(this, matcher);
+        }
+
+
+        /**
+         * build configuration.
+         *
+         * @return Plugin advice definition.
+         */
+        public PluginAdviceDefine install() {
+            return new PluginAdviceDefine(constructorPoints, instanceMethodPoints, classStaticMethodPoints);
+        }
+
+        /**
+         * Instance method intercepting point configuration builder.
+         */
+        public static final class InstanceMethodPointBuilder {
+            private final Builder builder;
+
+            private String classNameOfAdvice;
+
+            private boolean overrideArgs;
+
+            private ElementMatcher<? super MethodDescription> matcher;
+
+            private InstanceMethodPointBuilder(final Builder builder, final ElementMatcher<? super MethodDescription> matcher) {
+                this.builder = builder;
+                this.matcher = matcher;
+            }
+
+            /**
+             * to configure implementation for intercepting point.
+             *
+             * @param classNameOfAdvice the class name of advice
+             * @return Instance method point configurer.
+             */
+            public InstanceMethodPointBuilder implement(final String classNameOfAdvice) {
+                this.classNameOfAdvice = classNameOfAdvice;
+                return this;
+            }
+
+            /**
+             * to configure whether or not override the origin method arguments.
```

Comment:
to configure whether or not override the origin method arguments.  -- > Configure whether or not override the origin method arguments.